icu4c/source/i18n/messageformat2_parser.cpp

```diff
@@ -230,16 +254,31 @@ UnicodeSet* initNameChars(UErrorCode& status) {
 
 UnicodeSet* initTextChars(UErrorCode& status) {
     if (U_FAILURE(status)) {
-        return {};
+        return nullptr;
     }
 
+    initContentChars(status);
```

Comment:
now you are calling initContentChars multiple times, in this PR, once in initTextChars, a second time in initQuotedChars. Same issue in initWhitespace.

Reply:
Fixed